### PR TITLE
Fix: Add support for Roman Numerals to unique visitors grabber

### PR DESCRIPTION
**Pull request by martimavocado**
also added .isRoman() to stringutils
note: the new regex for tier detection only detects up to tier 10, this is fine for now but could break whenever hypixel adds new visitors

### CODE REVIEW

**Comment by hannibal002** [comment]
Maybe use `String.romanToDecimalIfNecessary()` instead?

**Review by hannibal002 — CHANGES_REQUESTED** [review]
see dms

File: src/main/java/at/hannibal2/skyhanni/features/garden/fortuneguide/CaptureFarmingGear.kt [hunk 1]
```diff
@@ -44,7 +44,7 @@ class CaptureFarmingGear {
     private val patternGroup = RepoPattern.group("garden.fortuneguide.capture")
     private val tierPattern by patternGroup.pattern(
         "uniquevisitors.tier",
-        "§7Progress to Tier (?<nextTier>\\d+):.*"
+        "§7Progress to Tier (?<nextTier>\\d+|I{1,3}|IV|V|VI{1,3}|IX|X):.*"
```
Comment (lunaynx) [inline src/main/java/at/hannibal2/skyhanni/features/garden/fortuneguide/CaptureFarmingGear.kt]:
```suggestion
        "§7Progress to Tier (?<nextTier>\\w+:.*"
```

**Review by lunaynx — CHANGES_REQUESTED** [review]

File: src/main/java/at/hannibal2/skyhanni/features/garden/fortuneguide/CaptureFarmingGear.kt (hunk 1 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/garden/fortuneguide/CaptureFarmingGear.kt]: this

**Review by hannibal002 — CHANGES_REQUESTED** [review]
The pattern does not detect for me.

Trying to debug `tierPattern` at CaptureFarmingGear line 151 gives me this:
Method threw 'java.lang.NoSuchFieldException' exception.

**Comment by martimavocado** [comment]
> The pattern does not detect for me.

try enabling "force local loading" in repopattern settings

File: src/main/java/at/hannibal2/skyhanni/features/garden/fortuneguide/CaptureFarmingGear.kt [hunk 2]
```diff
@@ -150,7 +150,7 @@ class CaptureFarmingGear {
             var tierProgress = -1
             for (line in item.getLore()) {
                 tierPattern.matchMatcher(line) {
-                    tier = group("nextTier").toInt() - 1
+                    tier = group("nextTier").romanToDecimalIfNecessary()
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/garden/fortuneguide/CaptureFarmingGear.kt]: Add the removed -1 back again. (tier = nextTier - 1)

**Review by hannibal002 — CHANGES_REQUESTED** [review]
one small but important thing

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**